### ISSUE TITLE
sanskari- DMRI reconstruction using cnn

### ISSUE THREAD

**Issue by ys1999** [hackfest19/submissions#23]
- DMRI reconstruction using cnn
- Diagnostic image reconstruction through undersampled data through CNN
- Team Name - sanskari
- Yash Sharda - ys1999
   Sanskar Shrivastava - sunscar
   Kashyap Kumar Patel - kashyapkansagara
   Naman gupta 
   Burhanuddin - Jonty0098
- **Demo Link**: _(if any, this might contain a website/ mobile application link/ short video, etc.)_
- https://github.com/ys1999/sanskari.git
## 🔥 Your Pitch
Image Reconstruction of Dynamic Magnetic Resonance Imaging of under-sampled data using CNN

Due to unavailability of the suitable dataset outright on the internet, we initially set up the layout of the model without any outright features to be extracted using Convulational Neural Network. And hence in the process defined the way CNN shall perceive images, read them, resize them and create an array of the time series of images. Then pre-processing of the images was carried out to be able to feed them to the model and get the required output.
The U-NET Architecture was seen as a suitable choice so as to explore the Brain MRI Images using Deep Learning Framework Keras. To extract the suitable features from the image, a conv. layer is implemented using Deep learning framework Keras which consisted of 32(3*3) filters that were to be convoluted with input images with stride 1 to extract the suitable features and perform Zero-padding(To maintain the dimension of image) as well as Batch Normalisation( To keep the our epochs in sweet range of our activation function). And A ReLU layer was used to add non-linearity in the model. A max-pool layer with dimension 2*2 is implemented to down-sample the input values to make values manageable to work with. Deconvolutional Layers were implemented so as to up-sample the retrieved data or to revert the image back to the original dimensions. Mean square Loss function was selected as the suitable loss function to minimise the loss of features. 

TIll next submission, we expect to have completely trained model with suitable dataset so as to have minimum loss and accuracy on the validation set is maximised. We shall have tested the robustness of our trained model by adding the noise to the test images and evaluate how well the model performs and finally to measure the performance of model or test our predictions we shall use quantitative peak signal to noise ratio.

Note: All the hyperparameters chosen are subjected to the choice of datasets and may change in order to maximise the accuracy on the validation dataset.

## 🔦 Any other specific thing you want to highlight?
_(Optional)_

## ✅ Checklist

**Before you post the issue**:
- [ ] You have followed the issue title format.
- [ ] You have made no other issue for this submission.
- [ ] You have provided all the information correctly.
